plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/extractor/PomParser.java

```diff
@@ -0,0 +1,182 @@
+package io.jenkins.tools.pluginmodernizer.core.extractor;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+public class PomParser {
+
+    private static final Logger logger = Logger.getLogger(PomParser.class.getName());
+
+    private final Document doc;
+    private final XPath xpath;
+    private final Map<String, String> properties;
+
+    public PomParser(String filePath) {
```

Comment:
OpenRewrite will do the parsing for you. I don't think that's the goal to implement a custom parser.